src/features/filters/components/filter-builder.tsx

```diff
@@ -292,22 +292,23 @@ function FilterBuilderForm({
                   ) : filter.type === "number" ||
                     filter.type === "numberObject" ? (
                     <Input
-                      value={filter.value?.toString() ?? ""}
+                      value={filter.value?.toString().replaceAll(",", ".") ?? ""}
                       type="number"
                       step="0.01"
-                      onChange={(e) =>
+                      lang="en_EN"
+                      onChange={(e) => {
+                        const { value } = e.target;
                         handleFilterChange(
                           {
                             ...filter,
                             value:
-                              isNaN(Number(e.target.value)) ||
-                              e.target.value.endsWith(".")
-                                ? e.target.value
-                                : Number(e.target.value),
+                              isNaN(Number(value)) || /^[.0]+$/.test(value)
```

Comment:
why do we need the regex check? The way i understand it, it checks for strings which consist of one or many `1` or `.` characters, correct?

Reply:
The purpose of the regex is to allow zero to be in putted after a decimal point, previously it wasn't functioning properly. Although we will be pushing a commit that covers more testcases for the regex.

Reply:
We have pushed an updated version of regex that should cover more testcases for the input field.